### PR TITLE
make: add Makefile config for common tasks

### PR DESCRIPTION
**Pull request by RandyMcMillan**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+#######################
+.PHONY: unix
+unix: mpy-cross
+	#bash -c "pushd external/micropython/mpy-cross &> /dev/null && make && popd &> /dev/null"
+	bash -c "pushd unix &> /dev/null && make setup ngu-setup && popd &> /dev/null"
+#######################
+.PHONY: simulator
+simulator: unix
+	bash -c "pushd unix &> /dev/null && make && ./simulator.py && popd &> /dev/null"
+#######################
```
Comment (RandyMcMillan) [inline Makefile:185]: PR #76

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+PWD 									?= pwd_unknown
+
+THIS_FILE								:= $(lastword $(MAKEFILE_LIST))
+export THIS_FILE
+TIME									:= $(shell date +%s)
+export TIME
+
+ARCH									:= $(shell uname -m)
+export ARCH
+
```
Comment (RandyMcMillan) [inline Makefile:12]: TODO: add arm64/aarch64 support

**Comment by doc-hex** [comment]
Before you get too far along here, I'm not too interested in making the Makefiles more complex than they absolutely need to be. I'd rather see the same energy on containerizing the Simulator, so it's easy to run w/o building. Building the binaries is already effectively cross-platform thanks to (gag) the Docker stuff.

**Comment by nvk** [comment]
+1 on complexity, this will hurt us in the future. KISS is best for Makefiles

+1 on efforts for Simulators being easier to use, if containers or signed binaries I’m indifferent.

**Pull request closed without merging**